src/render/node_collection.rs

```diff
@@ -0,0 +1,105 @@
+use crate::render::graph::Node;
+use std::cell::RefCell;
+use std::ops::{Index, IndexMut};
+
+pub struct NodeCollection {
+    nodes: Vec<Option<RefCell<Node>>>,
+}
+
+impl NodeCollection {
+    pub fn new() -> Self {
+        let mut instance = Self {
+            nodes: Vec::with_capacity(64),
```

Comment:
I wonder if we should be more agressive here and grab more memory at startup here, e.g. `Vec::with_capacity(2048)`

Reply:
`std::mem::size_of::<RefCell<Node>>()` is 160 bytes, so that would mean over 300kB allocated.
That's not terrible of course, but I think most audio graphs really don't need that many nodes.

Whatever the initial size, we do need to think about the performance impact of outgrowing that size.
- for the reallocation, we could perform it on the control thread because it will know when we run out of space
- for the memcpy, we should maybe look at reducing the side of `Node` #208 or an amortized growth like https://github.com/jonhoo/griddle